earth_enterprise/src/server/wsgi/wms/ogc/implementation/common.py

```diff
@@ -90,9 +90,12 @@ def GenerateOutputCommon(self):
       # Ignore the 0x at the beginning of the hex string.
       # Otherwise str.decode("hex") will throw
       # "TypeError: Non-hexadecimal digit found" error.
-      if bgcolor[:2] == "0x":
-        bgcolor = bgcolor[2:]
-      bgcolor = tuple(ord(c) for c in bgcolor.decode("hex"))
+      try:
+        if bgcolor[:2] == "0x":
+          bgcolor = bgcolor[2:]
+        bgcolor = tuple(ord(c) for c in bgcolor.decode("hex"))
+      except:
```

Comment:
This check on the BGCOLOR value is a nice addition. The BGCOLOR field is optional in a GetMap request, and I didn't see anything in the WMS implementation spec about returning an exception on bad BGCOLOR values. So it's better to set the value to the default and carry on with the request, as you've done here. I like it much better this way.

The bare except is really broad, though. We should specifically catch the TypeError mentioned above here.
- [ ] Narrow the except to catch only `TypeError`s 

It might also be worth noting in a comment for posterity that this whole conversion from a hex string looks different in Python 3 and raises a ValueError when it encounters problems.